src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -186,52 +211,142 @@ dumpBabelGUCs(Archive *fout)
-		cast->dobj.dump = DUMP_COMPONENT_NONE;
-}
+	switch (dobj->objType)
+	{
+		case DO_CAST:
+			{
+				CastInfo    	*cast = (CastInfo *) dobj;
+				TypeInfo    	*sTypeInfo;
+				TypeInfo    	*tTypeInfo;
+				ExtensionInfo	*ext = findOwningExtension(cast->dobj.catId);
+
+				/* Skip if cast is not a member of babelfish extension */
+				if (ext == NULL || strcmp(ext->dobj.name, "babelfishpg_common") != 0)
+					return;
+
+				sTypeInfo = findTypeByOid(cast->castsource);
+				tTypeInfo = findTypeByOid(cast->casttarget);
+
+				/*
+				* Do not dump following unused CASTS:
+				* pg_catalog.bool -> sys.bpchar
+				* pg_catalog.bool -> sys.varchar
+				*/
+				if (sTypeInfo && tTypeInfo &&
+						sTypeInfo->dobj.namespace &&
+						tTypeInfo->dobj.namespace &&
+						strcmp(sTypeInfo->dobj.namespace->dobj.name, "pg_catalog") == 0 &&
+						strcmp(tTypeInfo->dobj.namespace->dobj.name, "sys") == 0 &&
+						strcmp(sTypeInfo->dobj.name, "bool") == 0 &&
+						(strcmp(tTypeInfo->dobj.name, "bpchar") == 0 ||
+						strcmp(tTypeInfo->dobj.name, "varchar") == 0))
+					cast->dobj.dump = DUMP_COMPONENT_NONE;
+			}
+			break;
+		case DO_TABLE:
+			{
+				TableInfo *tbinfo = (TableInfo *) dobj;
+
+				if (fout->dopt->binary_upgrade)
+					return;
+
+				switch (tbinfo->relkind)
+				{
+					case RELKIND_VIEW:
+					{
+						/*
+						 * There is special case with sysdatabases view,
+						 * we will not dump this view only when it's in default
+						 * databases (master/msdb/tempdb), otherwise we
+						 * will always dump it.
+						 */
+						if ((strcmp(tbinfo->dobj.namespace->dobj.name, "master_dbo") == 0 ||
+							strcmp(tbinfo->dobj.namespace->dobj.name, "msdb_dbo") == 0 ||
+							strcmp(tbinfo->dobj.namespace->dobj.name, "tempdb_dbo") == 0) &&
+							strcmp(tbinfo->dobj.name, "sysdatabases") == 0)
+						{
+							tbinfo->dobj.dump = DUMP_COMPONENT_NONE;
+							break;
+						}
+
+						/* Just skip if it's a system view */
+						if (is_ms_shipped(dobj, fout))
+							tbinfo->dobj.dump = DUMP_COMPONENT_NONE;
+					}
+					break;
+					case RELKIND_SEQUENCE:
+					{
+						if (strcmp(dobj->namespace->dobj.name, "sys") == 0 &&
+							strcmp(dobj->name, "babelfish_db_seq") == 0)
+							dobj->dump &= ~DUMP_COMPONENT_ACL;
+					}
+					break;
+					default:
+						{
+							/*
+							 * Mark Babelfish catalog table data to be dumped if not in
+							 * binary-upgrade mode. This is needed since babelfish extensions
+							 * are not marked to be dumped so catalog table data explicitly
+							 * need to be marked as dumpable.
+							 */
+							if (isBabelfishConfigTable(tbinfo))
+								tbinfo->dobj.dump |= DUMP_COMPONENT_DATA;
+						}
+				}
+			}
+			break;
+		case DO_NAMESPACE:
+			{
+				NamespaceInfo *nsinfo = (NamespaceInfo *) dobj;
+
+				if (fout->dopt->binary_upgrade)
+					return;
+
+				/* Do not dump the definition of default babelfish schemas */
+				if (strncmp(nsinfo->dobj.name, "babelfishpg", 11) == 0 ||
```

Comment:
Yeah right, I guess this check is valid only for extensions only since as described in comment below, not dumping extension is going to be harmless but not dumping schema can be problematic.
The only reason to add this schema check is to exclude `CREATE SCHEMA` statement for `babelfish_telemetry` schema (which doesn't exist in OSS) so that there is no such error like:
```
ERROR:  schema "babelfishpg_telemetry" already exists
```
But now I have come to believe that getting this error is better than not dumping some user schema `babelfishxyz` so I have removed this schema check completely.